src/abstractdataframe/iteration.jl

```diff
@@ -79,7 +79,7 @@ Base.size(itr::DataFrameRows) = (size(parent(itr), 1), )
 
 Base.@propagate_inbounds Base.getindex(itr::DataFrameRows, i::Int) = parent(itr)[i, :]
 Base.@propagate_inbounds Base.getindex(itr::DataFrameRows, idx) =
-    eachrow(@view parent(itr)[idx, :])
+    eachrow(@view parent(itr)[idx isa AbstractVector{Int} ? copy(idx) : idx, :])
```

Comment:
Add a comment like this?
```
# `filter(f, ::AbstractArray)` in Base assumes that `idx` can be overwritten after calling `getindex`
```

Reply:
I do not think it is needed. This copying operation should be done anyway. The issue is that `array[idx]` operation should produce a result that does not change after `idx` is changed. So this PR is just a bugfix (that is why I marked it bug).

Reply:
OK. I'm not sure it's documented but we'll check that later.

BTW, detecting cases where we need to copy is potentially tricky. For example, when indexing with a `PooledArray{UInt}`, conversion to `AbstractArray{Int}` could result in a `PooledArray{Int}` which shares the same reference codes, only with a different pool. Currently we don't do it but that would make sense as an optimization. If we want to be super defensive we could do something like
```suggestion
    eachrow(@view parent(itr)[idx isa AbstractVector && !eltype(idx) <: Bool ? copy(idx) : idx, :])
```